pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -1070,6 +1071,21 @@ func (os *OpenStack) CreateVolume(name string, size int, tags *map[string]string
 	return vol.ID, err
 }
 
+func (os *OpenStack) GetDevicePath(diskId string) string {
+	files, _ := ioutil.ReadDir("/dev/disk/by-id/")
```

Comment:
@metral: thanks for looking at this, `GetDevicePath` is actually specific to vanilla OpenStack (`os *OpenStack`). It was part of the Cinder volume plugin, which I generalized to make it work with Rackspace as well: https://github.com/kubernetes/kubernetes/pull/22023/files#diff-dcc20522f7683cc42d2cd40a27c2c155L92

For Rackspace I did not find a way to deduct the device via volume-id from the device tree at all, so I am querying the Cinder API for the device, which works fine: https://github.com/kubernetes/kubernetes/pull/22023/files#diff-dcc20522f7683cc42d2cd40a27c2c155L92